google/cloud/spanner/internal/connection_impl_test.cc

```diff
@@ -626,6 +652,78 @@ TEST(ConnectionImplTest, CommitCommitFailure) {
   EXPECT_THAT(commit.status().message(), HasSubstr("uh-oh in Commit"));
 }
 
+TEST(ConnectionImplTest, CommitCommit_IdempotentTransientSuccess) {
+  auto mock = std::make_shared<spanner_testing::MockSpannerStub>();
+
+  auto db = Database("dummy_project", "dummy_instance", "dummy_database_id");
+  auto conn = MakeConnection(db, mock);
+  EXPECT_CALL(*mock, CreateSession(_, _))
+      .WillOnce(
+          Invoke([&db](grpc::ClientContext&,
+                       spanner_proto::CreateSessionRequest const& request) {
+            EXPECT_EQ(db.FullName(), request.database());
+            spanner_proto::Session session;
+            session.set_name("test-session-name");
+            return session;
+          }));
+  EXPECT_CALL(*mock, Commit(_, _))
+      .WillOnce(Invoke([](grpc::ClientContext&,
+                          spanner_proto::CommitRequest const& request) {
+        EXPECT_EQ("test-session-name", request.session());
+        EXPECT_EQ("test-txn-id", request.transaction_id());
+        return Status(StatusCode::kUnavailable, "try-again");
+      }))
+      .WillOnce(Invoke([](grpc::ClientContext&,
+                          spanner_proto::CommitRequest const& request) {
+        EXPECT_EQ("test-session-name", request.session());
+        EXPECT_EQ("test-txn-id", request.transaction_id());
+        spanner_proto::CommitResponse response;
+        *response.mutable_commit_timestamp() =
+            internal::ToProto(Timestamp{std::chrono::seconds(123)});
+        return response;
+      }));
+
+  auto txn = MakeReadWriteTransaction();
```

Comment:
Should we add a comment, like below, about why we're binding an id?

Reply:
Done.